src/main/java/org/broadinstitute/hellbender/tools/exome/segmentation/ClusteringGenomicHMMSegmenter.java

```diff
@@ -135,41 +143,31 @@ private void learn() {
         boolean converged = false;
         while (!converged && iteration++ < MAX_EM_ITERATIONS) {
             logger.info(String.format("Beginning iteration %d of learning.", iteration));
-            logger.info(String.format("Current values of hidden states: %s.", Arrays.toString(hiddenStateValues)));
-            logger.info(String.format("Current weights of hidden states: %s.", Arrays.toString(weights)));
             logger.info(String.format("Current memory length: %f bases.", memoryLength));
 
             final double oldMemoryLength = memoryLength;
             final double[] oldWeights = weights.clone();
-            final double[] oldHiddenStateValues = hiddenStateValues.clone();
+            final List<HIDDEN> oldHiddenStateValues = new ArrayList<>(hiddenStateValues);
             performEMIteration();
             converged = oldWeights.length == weights.length &&
                     Math.abs(oldMemoryLength - memoryLength) < MEMORY_LENGTH_CONVERGENCE_THRESHOLD &&
                     GATKProtectedMathUtils.maxDifference(oldWeights, weights) < CONVERGENCE_THRESHOLD &&
-                    GATKProtectedMathUtils.maxDifference(oldHiddenStateValues, hiddenStateValues) < CONVERGENCE_THRESHOLD;
+                    hiddenStateValuesHaveConverged(oldHiddenStateValues);
         }
         parametersHaveBeenLearned = true;
     }
 
+    protected abstract boolean hiddenStateValuesHaveConverged(final List<HIDDEN> oldHiddenStateValues);
+
     // update the model and the concentration parameter with a single EM step
     private void performEMIteration() {
-        logger.info("Performing E step via the forward-backward algorithm.");
         final ExpectationStep expectationStep = new ExpectationStep();
-        logger.info("Performing M step optimizations of parameters.");
-        logger.info("Relearning hidden state values.");
         relearnHiddenStateValues(expectationStep);
-        logger.info("Relearning hidden state weights.");
         relearnWeights(expectationStep);
-        logger.info("Relearning memory length.");
         relearnMemoryLength(expectationStep);
-        logger.info("Attempting larger change in memory length");
         attemptBigChangeInMemoryLength();
-        logger.info("Relearning additional parameters.");
         relearnAdditionalParameters(expectationStep);
-        logger.info("Attempting to reduce the number of hidden states used.");
         pruneUnusedComponents();
-        logger.info("Number of states after pruning: " + hiddenStateValues.length);
-        logger.info("Relearning concentration hyperparameter.");
```

Comment:
Output restored.  As for number of segments, that's a little trickier because iterations' E step's don't get segments -- they use forward-backward, not Viterbi.  If you feel strongly I could write some brief code to estimate a segmentation from the pointwise maximum E step posteriors.